### PR TITLE
Test GitHub actions

### PR DESCRIPTION
**Pull request by JavaDeveloper456788**

### CODE REVIEW

File: .github/workflows/create_tag.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+jobs:
+  tag_image:
+    name: Tag Image
+    runs-on: ubuntu-24.04
+    environment: dev
+
+    outputs:
+      ROUTE: ${{ steps.deploy-and-expose.outputs.route }}
+      SELECTOR: ${{ steps.deploy-and-expose.outputs.selector }}
+
```
Comment (cditcher) [inline .github/workflows/create_tag.yml:36]: Remove unused outputs

File: .github/workflows/deploy_test.yml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  MAX_CPU: "250m"
+  MIN_MEM: "500Mi"
+  MAX_MEM: "2Gi"
+  MIN_REPLICAS: "3"
+  MAX_REPLICAS: "5"
+  MAX_REPLICAS_TEST: '3'
+  HOST_ROUTE: '${{ secrets.SITE_URL }}'
+  CA_CERT: '${{ secrets.CA_CERT }}'
+  CERTIFICATE: '${{ secrets.CERT }}'
+  PRIVATE_KEY: '${{ secrets.PRIV_KEY }}'
```
Comment (cditcher) [inline .github/workflows/deploy_test.yml:31]: Remove unused constants

File: .github/workflows/deploy_test.yml [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+jobs:
+  deploy-to-openshift-test:
+    name: Deploy to OpenShift TEST
+    runs-on: ubuntu-24.04
+    environment: test
+
+    outputs:
+      ROUTE: ${{ steps.deploy-and-expose.outputs.route }}
+      SELECTOR: ${{ steps.deploy-and-expose.outputs.selector }}
+
```
Comment (cditcher) [inline .github/workflows/deploy_test.yml:47]: Remove unused outputs

File: .github/workflows/deploy_to_prod.yml [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  MIN_REPLICAS: "3"
+  MAX_REPLICAS: "5"
+  MAX_REPLICAS_TEST: '3'
+  BRANCH: master
+  HOST_ROUTE: '${{ secrets.SITE_URL }}'
+
+  CA_CERT: '${{ secrets.CA_CERT }}'
+  CERTIFICATE: '${{ secrets.CERT }}'
+  PRIVATE_KEY: '${{ secrets.PRIV_KEY }}'
+
```
Comment (cditcher) [inline .github/workflows/deploy_to_prod.yml:34]: Remove unused constants

File: .github/workflows/deploy_to_prod.yml [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+jobs:
+  deploy-to-openshift-prod:
+    name: Deploy to OpenShift PROD
+    runs-on: ubuntu-24.04
+    environment: prod
+
+    outputs:
+      ROUTE: ${{ steps.deploy-and-expose.outputs.route }}
+      SELECTOR: ${{ steps.deploy-and-expose.outputs.selector }}
+
```
Comment (cditcher) [inline .github/workflows/deploy_to_prod.yml:49]: Remove unused outputs

File: openshift/templates/cas-api/cas-api.dc.yaml [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    required: true
+  - name: CA_CERT
+    description: The CA Certificate
+    required: true
+  - name: CERTIFICATE
+    description: The Certificate
+    required: true
+  - name: PRIVATE_KEY
+    description: The private key
+    required: true
```
Comment (cditcher) [inline openshift/templates/cas-api/cas-api.dc.yaml:169]: Remove any unused parameters

File: openshift/templates/ecas-api/api.dc.yaml [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+    required: true
+  - name: CA_CERT
+    description: The CA Certificate
+    required: true
+  - name: CERTIFICATE
+    description: The Certificate
+    required: true
+  - name: PRIVATE_KEY
+    description: The private key
+    required: true
```
Comment (cditcher) [inline openshift/templates/ecas-api/api.dc.yaml:169]: Remove unused parameters

File: openshift/templates/ecas-frontend/frontend.dc.yaml [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+    required: true
+  - name: CA_CERT
+    description: The CA Certificate
+    required: true
+  - name: CERTIFICATE
+    description: The Certificate
+    required: true
+  - name: PRIVATE_KEY
+    description: The private key
+    required: true
```
Comment (cditcher) [inline openshift/templates/ecas-frontend/frontend.dc.yaml:170]: Remove unused parameters

File: openshift/templates/cas-api/cas-api.dc.yaml [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+  apiVersion: template.openshift.io/v1
+  kind: Template
+  labels:
+    template: "${REPO_NAME}-template"
+  metadata:
+    name: "${REPO_NAME}-cas-api-dc"
+    creationTimestamp:
+  objects:
+  - apiVersion: v1
+    kind: DeploymentConfig
```
Comment (cditcher) [inline openshift/templates/cas-api/cas-api.dc.yaml:11]: Replace with Deployment object: https://eccbc.atlassian.net/wiki/spaces/EDKB/pages/536969221/DeploymentConfig+to+Deployments

File: openshift/templates/ecas-api/api.dc.yaml [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+  apiVersion: template.openshift.io/v1
+  kind: Template
+  labels:
+    template: "${REPO_NAME}-template"
+  metadata:
+    name: "${REPO_NAME}-api-dc"
+    creationTimestamp:
+  objects:
+  - apiVersion: v1
+    kind: DeploymentConfig
```
Comment (cditcher) [inline openshift/templates/ecas-api/api.dc.yaml:11]: Replace with Deployment object: https://eccbc.atlassian.net/wiki/spaces/EDKB/pages/536969221/DeploymentConfig+to+Deployments

File: openshift/templates/ecas-frontend/frontend.dc.yaml [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+  kind: Template
+  labels:
+    template: "${REPO_NAME}-template"
+  metadata:
+    name: "${REPO_NAME}-frontend-dc"
+    creationTimestamp:
+  objects:
+  - apiVersion: v1
+    kind: DeploymentConfig
+    metadata:
```
Comment (cditcher) [inline openshift/templates/ecas-frontend/frontend.dc.yaml:12]: Replace with Deployment object: https://eccbc.atlassian.net/wiki/spaces/EDKB/pages/536969221/DeploymentConfig+to+Deployments

**Review by cditcher — CHANGES_REQUESTED** [review]
Hi Khaled, a bunch of comments to address. Thanks.

File: .github/workflows/build.from.master.branch.deploy.to.dev.yml [hunk 12]
```diff
@@ -0,0 +1,294 @@
...
+  CERTIFICATE: '${{ secrets.CERT }}'
+  PRIVATE_KEY: '${{ secrets.PRIV_KEY }}'
+  SOURCE_REPOSITORY: 'https://github.com/bcgov/EDUC-HUB.git'
+  DOTNET_STARTUP_PROJECT: 'CASInterfaceService/cas-interface-service.csproj'
+  IMAGE_REGISTRY: 'ghcr.io/${{ github.repository_owner }}'
+  IMAGE_REGISTRY_USER: '${{ github.actor }}'
+  IMAGE_REGISTRY_PASSWORD: '${{ github.token }}'
+  IMAGE_NAME_FRONTEND: ecas-frontend
+  IMAGE_NAME_API: ecas-api
+  IMAGE_NAME_CAS: cas-api
```
[the remark this answers is not in the document]
Comment (JavaDeveloper456788) [reply .github/workflows/build.from.master.branch.deploy.to.dev.yml]: Fixed

File: .github/workflows/build.from.master.branch.deploy.to.dev.yml [hunk 13]
```diff
@@ -0,0 +1,294 @@
+name: Build & Deploy to DEV from master branch
+env:
+  OPENSHIFT_SERVER: '${{ secrets.OPENSHIFT_SERVER }}'
+  OPENSHIFT_TOKEN: '${{ secrets.OPENSHIFT_TOKEN }}'
+  OPENSHIFT_NAMESPACE_DEV: '${{ secrets.ECAS_NAMESPACE_NO_ENV }}-dev'
+  OPENSHIFT_NAMESPACE_TEST: '${{ secrets.ECAS_NAMESPACE_NO_ENV }}-test'
+  REDHAT_REGISTRY_USERNAME: '${{ secrets.REDHAT_REGISTRY_USERNAME }}'
+  REDHAT_REGISTRY_PASSWORD: '${{ secrets.REDHAT_REGISTRY_PASSWORD }}'
+  CA_CERT: '${{ secrets.CA_CERT }}'
+  CERTIFICATE: '${{ secrets.CERT }}'
```
[the remark this answers is not in the document]
Comment (JavaDeveloper456788) [reply .github/workflows/build.from.master.branch.deploy.to.dev.yml:9]: it's used here :   oc process -f ./openshift/templates/ecas-frontend/frontend.dc.yaml -p APP_NAME=${{ env.APP_NAME }} -p REPO_NAME=${{ env.REPO_NAME }} -p BRANCH=${{ env.BRANCH }} -p NAMESPACE=${{ env.OPENSHIFT_NAMESPACE_DEV }} -p TAG=${{ env.TAG }} -p MIN_REPLICAS=${{ env.MIN_REPLICAS }} -p MAX_REPLICAS=${{ env.MAX_REPLICAS }} -p MIN_CPU=${{ env.MIN_CPU }} -p MAX_CPU=${{ env.MAX_CPU }} -p MIN_MEM=${{ env.MIN_MEM }} -p MAX_MEM=${{ env.MAX_MEM }} -p HOST_ROUTE=${{ env.HOST_ROUTE }} -p CA_CERT="${{ env.CA_CERT }}" -p CERTIFICATE="${{ env.CERTIFICATE }}" -p PRIVATE_KEY="${{ env.PRIVATE_KEY }}"\
          | oc apply -f -

File: .github/workflows/build.from.master.branch.deploy.to.dev.yml [hunk 14]
```diff
@@ -0,0 +1,294 @@
...
+    name: Build and deploy to DEV
+    runs-on: ubuntu-24.04
+    environment: dev
+    outputs:
+      ROUTE: '${{ steps.deploy-and-expose.outputs.route }}'
+      SELECTOR: '${{ steps.deploy-and-expose.outputs.selector }}'
+      frontend_image: '${{ steps.push-image-frontend.outputs.registry-path }}'
+      api_image: '${{ steps.push-image-api.outputs.registry-path }}'
+      cas_image: '${{ steps.push-image-api.outputs.registry-path }}'
+
```
[the remark this answers is not in the document]
Comment (JavaDeveloper456788) [reply .github/workflows/build.from.master.branch.deploy.to.dev.yml]: Fixed

File: .github/workflows/build.from.master.branch.deploy.to.dev.yml [hunk 15]
```diff
@@ -0,0 +1,294 @@
...
+            else {
+              core.info(`✅ All the required secrets are set`);
+            }
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: List files
+        run: ls -R
+
```
[the remark this answers is not in the document]
Comment (JavaDeveloper456788) [reply .github/workflows/build.from.master.branch.deploy.to.dev.yml:95]: Fixed

File: .github/workflows/create_tag.yml (hunk 1 above)
Comment (JavaDeveloper456788) [reply .github/workflows/create_tag.yml]: Fixed

File: .github/workflows/deploy_test.yml (hunk 2 above)
Comment (JavaDeveloper456788) [reply .github/workflows/deploy_test.yml:31]: They are used here :  oc process -f ./openshift/templates/ecas-frontend/frontend.dc.yaml -p APP_NAME=${{ env.APP_NAME }} -p REPO_NAME=${{ env.REPO_NAME }} -p BRANCH=${{ env.BRANCH }} -p NAMESPACE=${{ env.OPENSHIFT_NAMESPACE_TEST }} -p TAG=${{ steps.get-latest-tag.outputs.tag  }} -p MIN_REPLICAS=${{ env.MIN_REPLICAS }} -p MAX_REPLICAS=${{ env.MAX_REPLICAS }} -p MIN_CPU=${{ env.MIN_CPU }} -p MAX_CPU=${{ env.MAX_CPU }} -p MIN_MEM=${{ env.MIN_MEM }} -p MAX_MEM=${{ env.MAX_MEM }} -p HOST_ROUTE=${{ env.HOST_ROUTE }} -p CA_CERT="${{ env.CA_CERT }}" -p CERTIFICATE="${{ env.CERTIFICATE }}" -p PRIVATE_KEY="${{ env.PRIVATE_KEY }}"\
          | oc apply -f -

File: .github/workflows/deploy_test.yml (hunk 3 above)
Comment (JavaDeveloper456788) [reply .github/workflows/deploy_test.yml:45]: Fixed

File: .github/workflows/deploy_to_prod.yml (hunk 5 above)
Comment (JavaDeveloper456788) [reply .github/workflows/deploy_to_prod.yml:46]: Fixed

File: .github/workflows/build.from.master.branch.deploy.to.dev.yml (hunk 13 above)
[the remark this answers is not in the document]
Comment (JavaDeveloper456788) [reply .github/workflows/build.from.master.branch.deploy.to.dev.yml]: Fixed

File: .github/workflows/deploy_test.yml (hunk 2 above)
Comment (JavaDeveloper456788) [reply .github/workflows/deploy_test.yml]: Fixed

File: openshift/templates/ecas-api/api.dc.yaml (hunk 7 above)
Comment (JavaDeveloper456788) [reply openshift/templates/ecas-api/api.dc.yaml:169]: Fixed

File: openshift/templates/ecas-frontend/frontend.dc.yaml (hunk 8 above)
Comment (JavaDeveloper456788) [reply openshift/templates/ecas-frontend/frontend.dc.yaml]: Fixed

File: openshift/templates/cas-api/cas-api.dc.yaml (hunk 6 above)
Comment (JavaDeveloper456788) [reply openshift/templates/cas-api/cas-api.dc.yaml]: Fixed

**Pull request merged**